src/pages/DelegationPage/components/DelegateTokensModal.vue

```diff
@@ -0,0 +1,265 @@
+          {{ $t('delegate-tokens-modal.delegate-text') }}
+        </p>
+        <p class="delegate-tokens-modal__available-amount">
+          {{ `${MOCKED_MOR_AMOUNT} ${web3ProvidersStore.rewardsTokenSymbol}` }}
+        </p>
+      </div>
+      <div class="delegate-tokens-modal__inputs-wrp">
+        <input-field
+          v-model="form.address"
+          class="delegate-tokens-modal__input-field"
+          :placeholder="$t('delegate-tokens-modal.wallet-input-placeholder')"
+          :error-message="getFieldErrorMessage('address')"
+          :disabled="isSubmitting || Boolean(delegateAddress)"
+          @blur="touchField('address')"
+        />
+        <input-field
+          v-model="form.amount"
+          class="delegate-tokens-modal__input-field"
+          :placeholder="
+            $t('delegate-tokens-modal.mor-input-placeholder', {
+              asset: web3ProvidersStore.rewardsTokenSymbol,
+            })
+          "
+          :error-message="getFieldErrorMessage('amount')"
+          :disabled="isSubmitting"
+          @blur="touchField('amount')"
+        >
+          <template #nodeRight>
+            <app-button
+              class="delegate-tokens-modal__input-field-btn"
+              scheme="link"
+              text="max"
+              :disabled="isSubmitting || !Number(depositTokenBalance)"
+              @click="inputMaxTokensAmount"
+            />
+          </template>
+        </input-field>
+        <select-field
+          v-model="form.delegationRights"
+          class="delegate-tokens-modal__input-field"
+          :value-options="delegationOptions"
+          :placeholder="$t('delegate-tokens-modal.rights-input-placeholder')"
+          :error-message="getFieldErrorMessage('delegationRights')"
+          :disabled="isSubmitting"
+          @blur="touchField('delegationRights')"
+        />
+      </div>
+      <div class="delegate-tokens-modal__buttons-wrp">
+        <app-button
+          class="delegate-tokens-modal__btn"
+          color="secondary"
+          :text="$t('delegate-tokens-modal.cancel-button')"
+          @click="closeModal"
+        />
+        <app-button
+          class="delegate-tokens-modal__btn"
+          :text="$t('delegate-tokens-modal.confirm-button')"
+          :disabled="isSubmitting || !isFieldsValid"
+          @click="submit"
+        />
+      </div>
+    </div>
+  </basic-modal>
+</template>
+
+<script lang="ts" setup>
+import { AppButton, BasicModal } from '@/common'
+import { computed, reactive, ref } from 'vue'
+import { config } from '@config'
+import { InputField, SelectField } from '@/fields'
+import { useFormValidation, useI18n } from '@/composables'
+import { address, required, minEther, maxEther } from '@/validators'
+import { useWeb3ProvidersStore } from '@/store'
+import { bus, BUS_EVENTS, ErrorHandler, getEthExplorerTxUrl } from '@/helpers'
+import { DELEGATION_RIGHTS } from '@/enums'
+
+const MIN_DELEGATION_AMOUNT = 0.000001
```

Comment:
form fields validation consts is better to store in one place in case you'll ever need to reuse them.

example:
```ts
export const FIELD_LENGTH = {
    contractName: {
        min: 3,
        max: 64,
    },
    addressList: {
        min: 1,
        max: Number.MAX_SAFE_INTEGER,
    },
    setName: {
        min: 1,
        max: 100,
    },
    description: {
        min: 0,
        max: 300,
    },
} as const

export const FIELD_VALUE = {
    tokenThreshold: {
        min: 1,
        default: 50_000,
        max: Number.MAX_SAFE_INTEGER,
    },
} as const
```